pkg/cmd/cli/cmd/buildlogs.go

```diff
@@ -59,39 +54,9 @@ func NewCmdBuildLogs(fullName string, f *clientcmd.Factory, out io.Writer) *cobr
 }
 
 // RunBuildLogs contains all the necessary functionality for the OpenShift cli build-logs command
-func RunBuildLogs(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, opts api.BuildLogOptions, args []string) error {
+func RunBuildLogs(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, opts api.BuildLogOptions, args []string) error {
 	if len(args) != 1 {
-		// maximum time to wait for a list of builds
-		timeout := 800 * time.Millisecond
-		// maximum number of builds to list
-		maxBuildListLen := 10
-		ch := make(chan error)
-		go func() {
-			// TODO fetch via API no more than maxBuildListLen builds
-			builds, err := getBuilds(f)
-			if err != nil {
-				return
-			}
-			if len(builds) == 0 {
-				ch <- cmdutil.UsageError(cmd, "There are no builds in the current project")
-				return
-			}
-			sort.Sort(sort.Reverse(api.ByCreationTimestamp(builds)))
-			msg := "A build name is required. Most recent builds:"
-			for i, b := range builds {
-				if i == maxBuildListLen {
-					break
-				}
-				msg += fmt.Sprintf("\n* %s\t%s\t%s ago", b.Name, b.Status.Phase, units.HumanDuration(time.Since(b.CreationTimestamp.Time)))
-			}
-			ch <- cmdutil.UsageError(cmd, msg)
-		}()
-		select {
-		case <-time.After(timeout):
-			return cmdutil.UsageError(cmd, "A build name is required")
-		case err := <-ch:
-			return err
-		}
+		return cmdutil.UsageError(cmd, "A build name is required - you can run `%s get builds` to list builds", fullName)
```

Comment:
Only if project != default namespace.  I don't think users should be using
-n in day to day.

On Fri, Aug 7, 2015 at 11:09 AM, Ben Parees <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/buildlogs.go
> <https://github.com/openshift/origin/pull/4044#discussion_r36528108>:
>
> > -			msg := "A build name is required. Most recent builds:"
> > -			for i, b := range builds {
> > -				if i == maxBuildListLen {
> > -					break
> > -				}
> > -				msg += fmt.Sprintf("\n* %s\t%s\t%s ago", b.Name, b.Status.Phase, units.HumanDuration(time.Since(b.CreationTimestamp.Time)))
> > -			}
> > -			ch <- cmdutil.UsageError(cmd, msg)
> > -		}()
> > -		select {
> > -		case <-time.After(timeout):
> > -			return cmdutil.UsageError(cmd, "A build name is required")
> > -		case err := <-ch:
> > -			return err
> > -		}
> > +		return cmdutil.UsageError(cmd, "A build name is required - you can run `%s get builds` to list builds", fullName)
>
> customize the message/sample command to include the namespace they used
> when they invoked oc build-logs
>
> eg "oc get builds -n "
>
> if a project was specified.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4044/files#r36528108>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift